src/Concordium/Client/Types/Contract/Schema.hs

```diff
@@ -94,8 +94,32 @@ data ModuleSchema
   deriving (Eq, Show, Generic)
 
 -- |Create a getter based on the wasm version.
+-- Will first attempt to parse module schema as versioned and if it fails,
+-- it will attempt parsing based on the wasmVersion.
 getModuleSchema :: Wasm.WasmVersion -> S.Get ModuleSchema
-getModuleSchema wasmVersion = S.label "ModuleSchema" $
+getModuleSchema wasmVersion = S.label "ModuleSchema" $ do
+  prefix :: Word16 <- S.lookAhead S.get
+  if prefix /= maxBound then
+    getVersionedModuleSchema
+  else
+    getUnversionedModuleSchema wasmVersion
+
+-- |Getter for module schema with magic prefix and version.
+getVersionedModuleSchema :: S.Get ModuleSchema
+getVersionedModuleSchema = S.label "ModuleSchema" $ do
+  prefix :: Word16 <- S.get
+  if prefix /= maxBound then
+    fail "Schema is not prefixed with two full bytes"
```

Comment:
Consider adding some information about the fact that it tried parsing a versioned module schema, but the bytes didn't have the correct magic prefix, so it isn't a _versioned_ schema.

Reply:
I have added to the error message